airlock/business_logic.py

```diff
@@ -1350,28 +1357,35 @@ def withdraw_file_from_request(
         release_request.set_filegroups_from_dict(filegroup_data)
         return release_request
 
-    def release_files(self, request: ReleaseRequest, user: User):
-        """Release all files from a request to job-server.
+    def release_files(self, release_request: ReleaseRequest, user: User):
+        """Release all files from a release_request to job-server.
 
         This currently uses the old api, and is shared amongst provider
         implementations, but that will likely change in future.
         """
 
         # we check this is valid status transition *before* releasing the files
-        self.check_status(request, RequestStatus.RELEASED, user)
+        self.check_status(release_request, RequestStatus.RELEASED, user)
 
-        file_paths = request.get_output_file_paths()
+        file_paths = release_request.get_output_file_paths()
         self.validate_file_types(file_paths)
 
-        filelist = old_api.create_filelist(file_paths, request)
+        filelist = old_api.create_filelist(file_paths, release_request)
         jobserver_release_id = old_api.create_release(
-            request.workspace, filelist.json(), user.username
+            release_request.workspace, filelist.json(), user.username
         )
 
         for relpath, abspath in file_paths:
             old_api.upload_file(jobserver_release_id, relpath, abspath, user.username)
```

Comment:
I think ideally we'd reverse the order of operations here because, as things stand, a failure at just the wrong point in the process would lead to an uploaded file with no associated audit event. If we create the audit event first we risk having an event with no uploaded file, but I think that's a safer failure mode.

Reply:
yep done